File: 813-all-paths-from-source-to-target/all-paths-from-source-to-target.py
class Solution:
    def allPathsSourceTarget(self, graph: List[List[int]]) -> List[List[int]]:
        result = []
        path = []

        def dfs(node):
            path.append(node)
            if node == len(graph) - 1:
                result.append(list(path))
            else:
                for neighbours in graph[node]:
                    dfs(neighbours)
            path.pop()
        dfs(0)
        return result

File: 813-all-paths-from-source-to-target/all-paths-from-source-to-target.cpp
class Solution {
public:
    vector<vector<int>> allPathsSourceTarget(vector<vector<int>>& graph) {
        vector<int> path;
        vector<vector<int>> result;
        dfs(0, graph, path, result);
        return result;

    }
    void dfs(int node, vector<vector<int>> &graph, vector<int> &path, vector<vector<int>> &result){
        path.push_back(node);
        if(node == graph.size() - 1) result.push_back(path);
        else{
            for(auto neighbour : graph[node]){
                dfs(neighbour, graph, path, result);
            }
        }
        path.pop_back();
    }
};